pkg/frontend/adminactions/resources_list.go

```diff
@@ -44,35 +59,56 @@ func (a *azureActions) ResourcesList(ctx context.Context) ([]byte, error) {
 			a.log.Warnf("API version not found for type %q", *res.Type)
 			continue
 		}
+
+		if hasWritten {
+			writer.Write(commaAsByteArray)
+		}
+		hasWritten = true
+
 		switch *res.Type {
 		case "Microsoft.Compute/virtualMachines":
 			vm, err := a.virtualMachines.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				a.writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			a.writeObject(writer, arm.Resource{
 				Resource: vm,
 			})
 		default:
 			gr, err := a.resources.GetByID(ctx, *res.ID, apiVersion)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				a.writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			a.writeObject(writer, arm.Resource{
 				Resource: gr,
 			})
 		}
 	}
 
-	return json.Marshal(armResources)
+	_, err = writer.Write([]byte("]"))
+
+	return err
+}
+
+func (a *azureActions) writeObject(writer io.Writer, resource arm.Resource) {
+	bytes, err := resource.MarshalJSON()
+	if err != nil {
+		a.log.Warn(err) //very unlikely , only a handful of cases trigger an error
+		//here. and since we get the object from a database , it probably will never happen
+	} else {
```

Comment:
imo this is fine to use else, but we could reduce nesting by adding a return after logging on line 104 and then moving the code in the else block outside of the err check condition. 

```go
bytes, err := resource.MarshalJSON()
if err != nil {
  a.log.Warn(err)  // comment
  return
}

_, err = writer.Write(bytes)
if err != nil {
  a.log.Warn(err) //comment
}